### DISCUSSION TITLE
Using GitHub Release Feature

### DISCUSSION THREAD

**Discussion by oceanofthelost** [sierrafoxtrot/srecord#34] [Ideas]
Wanted to open a dialog on using GitHub release. 

Since Srecord is now using GitHub, should we use Release feature? GitHub release could be used to replace current [SourceForge Release](https://sourceforge.net/projects/srecord/files/) setup.

**Comment by sierrafoxtrot** [comment]
I'm using GitHub as it is nice for collaboration and some of the CI and related features that @jtxa has been introducing recently are pretty handy. However, I haven't considered moving away from SourceForge. I have to admit that the more we become dependent on the GitHub platform, the more I feel locked in. 

What value would it offer?

**Reply by oceanofthelost** [reply]
The fear of being locked in is rational, it is never great to be tied to one service. A benefit of using GitHub, GitLab, or BitBucket, is ease of collaboration, bug tracking, and feature planning, While all these features existed with Sourceforge, it is not as intuitive. 

For me, the benefit of using GitHub in this case would be to mirror Sourceforge. Knowing others in my generation and students I teach, many are turned off by Sourceforge as it is "old school" or "the website if too old looking", to "Shy use SVN when we have GIT"?. To cater to this group of individuals, which is growing, it would be useful to include released software. 

Maybe we should start simple and have GitHub auto-generate a release using actions using [automatic-releases](https://github.com/marketplace/actions/automatic-releases).

**Reply by jtxa** [reply]
GitHub was my choice, because they have free Runners for CI. And the MR on SF looked a bit ... pure.
And if you want more contributors (not just some enthusiasts), then this is the easiest way.

Having an automatic release build, I expect that it's easy to publish on GH. And it should be fairly easy to mirror them on SF (I hope for some ready-to-use action).

My priority was to establish a CI, before contributions (including mine) are coming. Next I want to fix warnings in C++ and improve CMake. Later also convert docs to `.rst`, but I have seen that @oceanofthelost already started with that (that's great).
Personally I don't want to spend my time ATM to find out how releases and their workflows on GH work. I assume that you need to setup a play-repo to test it out first.

**Reply by oceanofthelost** [reply]
I am playing around with `.rst` and seeing how well they render in one of my [branches](https://github.com/oceanofthelost/srecord/tree/documentation/change-documentation-format). 

I have also tested releases on GH. Within my [respiratory](https://github.com/oceanofthelost/srecord), the right hand side has a release section:

![image](https://user-images.githubusercontent.com/1684123/214470500-78547b69-6375-43e3-a3e8-70c76efe0a21.png)

I created releases for some SRecord releases. The workflow is simple:
1. Create a tag for a specific commit
2. Go to release page and select create release, and assign the tag from 1 to the release. 
3. Fill in changelog
4. Click submit
